ARCs/arc-0060.md

```diff
@@ -152,298 +182,206 @@ export interface HDWalletMetadata {
 
-A valid `StdDataStr` object for `ScopeType=AUTH` should look like the following:
+```ts
+export type ValidateFunction = {
+    structuredData: T
+} => boolean;
+```
 
-```json
-{
-  "ARC60Domain" : "arc60",
-  "bytes" : "[99, 50, 97, 57, 51, 99, ..., 57, 49, 53]"
-}
+#### Function `serialize`
+
+Function `serialize` flattens an instance of the structured data `T` into a byte array that can be fed into a  signing function. It is important that this function is deterministic, meaning that two parties that serialize the same data receive the same byte array. This ensures that given an instance of `T`, a signer and a verifier operate on the same byte string.
+
+```ts
+export type SerializeFunction = {
+    structuredData: T
+} => Uint8Array;
 ```
 
-#### Error interface `SignDataError`
+#### Scope `MX_RANDOM`
 
-The `SignDataError` object extends the `SignTxnsError` defined in [ARC-1](./arc-0001.md).
+This scope is inspired by the `signBytes` function in the Algorand SDK which is currently used by some applications. The `signBytes` function allows arbitrary data signing with the only restriction that the data is neither a valid transaction nor a logic signature. This is achieved by enforcing that the first two bytes are the domain separator `MX`. Apart from this restriction, the data is an opaque, unstructured byte string.
+
+Functions `parse`, `validate`, and `serialize` can be defined as follows:
 
 ```ts
-export interface SignDataError extends Error {
-  code: number;
-  data?: any;
-  failingSignData: (StdSigData | null);
+function parse(data: string): Uint8Array {
+    return Buffer.from(data, 'base64');
+}
+function validate(data: Uint8Array): boolean {
+    return data.length >= 2 && data[0] === 0x4D && data[1] === 0x58;
+}
+function serialize(data: Uint8Array): Uint8Array {
+    return data;
 }
 ```
 
-`SignDataError` uses the same error codes of `SignTxnsError` as well as the following codes:
+Since the data is an opaque byte string, wallets cannot inform the user about the actual content that the user is asked to sign. Therefore, it is recommended that applications propose scopes with richer schemas that allow wallets to present more useful information to users.
 
-| Status Code | Name | Description |
-| --- | --- | --- |
-| 4600 | Invalid scope | The scope of the signing action is not recognized by the wallet |
-| 4601 | Invalid schema | The schema does not comply with ARC-60 requirements |
-| 4602 | Invalid encoding | Data cannot be decoded with the given encoding |
+#### EXAMPLE Scope `DOG_OWNERSHIP`
 
-### Semantic Requirements
+This scope is intended purely as an example to showcase how a scope with a stricter schema can be signed. The `DOG_OWNERSHIP` scope signs a JSON object that states that a person owns a dog:
```

Comment:
The problem I see with this approach is that we are not using JSON schema but JSON objects. 

Schemas have a particular structure (https://json-schema.org/understanding-json-schema/reference) that we should follow, e.g. to enforce data validation rules against the schemas.